File: Leetcode/Array/C++/twoSum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int, int> hash_map;
        vector<int> result;

        if (nums.size() == 0) {
            return {};
        }
        for (int i = 0; i < nums.size(); i++) {
            hash_map[nums[i]] = i;
        }

        for (int i = 0; i < nums.size(); i++) {
            int val = target - nums[i];
            if (hash_map.find(val) != hash_map.end() && hash_map[val] != i) {
                result.push_back(i);
                result.push_back(hash_map[val]);
                break;
            }
        }

        return result;
    }
};

File: Leetcode/Array/Python/two_sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        hash_map = {}
        for indice, value in enumerate(nums):
            hash_map[value] = indice
            
        for indice, value in enumerate(nums):
            if target - value in hash_map:
                if indice != hash_map[target-value]:
                    return indice, hash_map[target-value]
            
        
